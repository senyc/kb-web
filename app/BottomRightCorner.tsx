interface BottomRightCornerProps {
  height: number;
  width: number;
}

const BottomRightCorner = ({ height, width }: BottomRightCornerProps) => {
  return (
    <svg
      className='absolute bottom-0 right-0'
      width={width}
      height={height}
    >
      <defs>
        <linearGradient
          id="grad1"
          x1="0%"
          y1="0%"
          x2="100%"
          y2="100%"
        >
          <stop offset="0" className="gradient2" />
          <stop offset=".5" className="gradient1" />
          <stop offset="1" className="gradient2" />
        </linearGradient>
      </defs>
      <path
        fill="url('#grad1')"
        d={`M292 300H0l20-20h255a5 5 0 0 0 5-5V20l20-20v292a8 8 0 0 1-8 8Z`}
      />
    </svg>
  );
};

export default BottomRightCorner;
