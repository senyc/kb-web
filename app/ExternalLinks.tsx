import Email from "./icons/Email";
import GitHub from "./icons/GitHub";
import LinkedIn from "./icons/LinkedIn";

export default function ExternalLinks() {
  return (
    <div className='absolute right-0 top-0 flex space-x-4'>
      <GitHub />
      <LinkedIn />
      <Email />
    </div>
  );
};
