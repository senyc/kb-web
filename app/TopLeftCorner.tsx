interface TopLeftCornerProps {
  height: number;
  width: number
  thickness?: number;
}

const TopLeftCorner = ({ height, width, thickness=20 }: TopLeftCornerProps) => {
  return (
    <svg 
      className='absolute top-0 left-0'
      width={width}
      height={height}
    >
      <defs>
        <linearGradient 
          id="grad1" 
          x1="0%" 
          y1="0%" 
          x2="100%" 
          y2="100%">
          <stop offset="0" className="gradient1" />
          <stop offset=".5" className="gradient2" />
          <stop offset="1" className="gradient1" />
        </linearGradient>
      </defs>
      <path 
        fill="url('#grad1')" d={`M8 0h${width - 8}l-20 ${thickness}H${thickness + 5}a5 5 0 0 0-5 5v${height - 45}L0 ${width}V8a8 8 0 0 1 8-8Z`} 
      />
    </svg>
  );
};

export default TopLeftCorner;  
