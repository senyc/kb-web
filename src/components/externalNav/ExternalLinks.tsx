'use client';

import Email from './icons/Email';
import GitHub from './icons/GitHub';
import LinkedIn from './icons/LinkedIn';

export default function ExternalLinks() {
  return (
    <>
      <GitHub />
      <Email />
      <LinkedIn />
    </>
  );
};
