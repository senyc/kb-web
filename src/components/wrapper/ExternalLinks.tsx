import Email from './icons/Email';
import GitHub from './icons/GitHub';
import LinkedIn from './icons/LinkedIn';
import Resume from './icons/Resume';

export default function ExternalLinks() {
  return (
    <div className='absolute right-3 top-3 flex flex-col space-y-3 lg:right-2 lg:top-2 lg:row-end-auto lg:flex-row lg:space-x-4 lg:space-y-0'>
      <Resume />
      <GitHub />
      <LinkedIn />
      <Email />
    </div>
  );
};
