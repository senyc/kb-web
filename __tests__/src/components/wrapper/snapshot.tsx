import { create } from 'react-test-renderer';

import BottomRightCorner from '@components/wrapper/BottomRightCorner';
import ExternalLinks from '@components/wrapper/ExternalLinks';
import TopLeftCorner from '@components/wrapper/TopLeftCorner';

describe('main elements match recorded snapshots', () => {
  describe('BottomRightCorner', () => {
    it('renders with standard size', () => {
      const tree = create(<BottomRightCorner width={300} height={300} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('TopLeftCorner', () => {
    it('renders with standard size', () => {
      const tree = create(<TopLeftCorner width={300} height={300} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('External Links', () => {
    it('exist upon render', () => {
      const tree = create(<ExternalLinks />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
