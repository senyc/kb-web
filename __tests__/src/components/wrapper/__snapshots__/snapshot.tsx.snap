// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main elements match recorded snapshots BottomRightCorner renders with standard size 1`] = `
<svg
  className="invisible absolute bottom-0 right-0 z-10 md:visible"
  height={300}
  width={300}
>
  <defs>
    <linearGradient
      id="grad1"
      x1="0%"
      x2="100%"
      y1="0%"
      y2="100%"
    >
      <stop
        className="gradient2"
        offset="0"
      />
      <stop
        className="gradient1"
        offset=".5"
      />
      <stop
        className="gradient2"
        offset="1"
      />
    </linearGradient>
  </defs>
  <path
    d="M292 300H0l20-20h255a5 5 0 0 0 5-5V20l20-20v292a8 8 0 0 1-8 8Z"
    fill="url('#grad1')"
  />
</svg>
`;

exports[`main elements match recorded snapshots External Links exist upon render 1`] = `
[
  <a
    className="duration-150 ease-in hover:scale-110"
    href="https://github.com/senyc"
    rel="me"
    target="_blank"
    title="GitHub"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
      />
    </svg>
  </a>,
  <a
    className="duration-150 ease-in hover:scale-110"
    href="mailto:kdbomhof@gmail.com"
    rel="me"
    target="_blank"
    title="Email"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
      />
      <polyline
        points="22,6 12,13 2,6"
      />
    </svg>
  </a>,
  <a
    className="duration-150 ease-in hover:scale-110"
    href="https://www.linkedin.com/in/kyler-bomhof/"
    rel="me"
    target="_blank"
    title="Linkedin"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
      />
      <rect
        height="12"
        width="4"
        x="2"
        y="9"
      />
      <circle
        cx="4"
        cy="4"
        r="2"
      />
    </svg>
  </a>,
]
`;

exports[`main elements match recorded snapshots TopLeftCorner renders with standard size 1`] = `
<svg
  className="invisible absolute left-0 top-0 z-10 sm:visible"
  height={300}
  width={300}
>
  <defs>
    <linearGradient
      id="grad1"
      x1="0%"
      x2="100%"
      y1="0%"
      y2="100%"
    >
      <stop
        className="gradient1"
        offset="0"
      />
      <stop
        className="gradient2"
        offset=".5"
      />
      <stop
        className="gradient1"
        offset="1"
      />
    </linearGradient>
  </defs>
  <path
    d="M8 0h292l-20 20H25a5 5 0 0 0-5 5v255L0 300V8a8 8 0 0 1 8-8Z"
    fill="url('#grad1')"
  />
</svg>
`;
